impl/src/main/java/org/arquillian/extension/governor/impl/GovernorTestClassScanner.java

```diff
@@ -61,6 +67,12 @@
     @Inject
     private Instance<GovernorConfiguration> governorConfiguration;
 
+    public void onBeforeSuite(@Observes BeforeSuite event) {
+        if (closePassedDecider.get() == null) {
```

Comment:
what if close passed decider is not null here? it means that it will be used in another "suite" cycle? what  about setting this to new impl every time? something like closePassedDecider.get() = null; closePassedDecider.set(new impl);